crypto/ec/curve25519.c

```diff
@@ -5715,3 +5715,38 @@ ossl_x25519_public_from_private(uint8_t out_public_value[32],
 
     OPENSSL_cleanse(e, sizeof(e));
 }
+
+void
+ossl_x25519_private_from_ed25519(uint8_t out_private_key[32],
+                                 const uint8_t private_key[32])
+{
+    uint8_t md[SHA512_DIGEST_LENGTH];
+
+    SHA512(private_key, 32, md);
+
+    md[0] &= 248;
+    md[31] &= 127;
+    md[31] |= 64;
+
+    memcpy(out_private_key, md, 32);
+
+    OPENSSL_cleanse(md, sizeof(md));
+}
+
+void
+ossl_x25519_public_from_ed25519(uint8_t out_public_value[32],
```

Comment:
sorry, it just occured to me, theres no documentation for these functions (in fact theres no documentation for any of the functions in this api).  That said, we're starting to undertake an effort to document our code using doxygen:
https://github.com/openssl/technical-policies/pull/94

Its not approved yet, so its not yet a requirement, but if you were inclined, would you be willing to add some doxygen style comments to the functions you added here, in the style suggested by the PR above?

No worries if you don't have time, but it would start us moving down the path of getting better documentation in our code base.